src/tqec/block/library/open.py

```diff
@@ -0,0 +1,262 @@
+        1: _with_measurements_on_data_qubits_on_side(
+            xx_memory_plaquette(PlaquetteOrientation.LEFT),
+            PlaquetteSide.DOWN,
+        ),
+        2: zzzz_memory_plaquette(),
+        3: _with_measurements_on_data_qubits_on_side(
+            xxxx_memory_plaquette(),
+            PlaquetteSide.DOWN,
+        ),
+        6: xxxx_memory_plaquette(),
+        7: zzzz_memory_plaquette(),
+        8: xx_memory_plaquette(PlaquetteOrientation.RIGHT),
+    }
+    return StandardComputationBlock(
+        QubitHorizontalBorders(dimension),
+        TemporalPlaquetteSequence(
+            initial_plaquettes, repeating_plaquettes, final_plaquettes
+        ),
+    )
+
+
+def zox_block(dimension: LinearFunction) -> StandardComputationBlock:
+    initial_plaquettes = defaultdict(empty_square_plaquette) | {
+        1: _with_resets_on_data_qubits_on_side(
+            zz_memory_plaquette(PlaquetteOrientation.LEFT),
+            PlaquetteSide.DOWN,
+            reset_basis=ResetBasis.X,
+        ),
+        2: xxxx_memory_plaquette(),
+        3: _with_resets_on_data_qubits_on_side(
+            zzzz_memory_plaquette(), PlaquetteSide.DOWN, reset_basis=ResetBasis.X
+        ),
+        6: zzzz_memory_plaquette(),
+        7: xxxx_memory_plaquette(),
+        8: zz_memory_plaquette(PlaquetteOrientation.RIGHT),
+    }
+    repeating_plaquettes = RepeatedPlaquettes(
+        defaultdict(empty_square_plaquette)
+        | {
+            1: zz_memory_plaquette(PlaquetteOrientation.LEFT),
+            2: xxxx_memory_plaquette(),
+            3: zzzz_memory_plaquette(),
+            6: zzzz_memory_plaquette(),
+            7: xxxx_memory_plaquette(),
+            8: zz_memory_plaquette(PlaquetteOrientation.RIGHT),
+        },
+        dimension,
+    )
+    final_plaquettes = defaultdict(empty_square_plaquette) | {
+        1: _with_measurements_on_data_qubits_on_side(
+            zz_memory_plaquette(PlaquetteOrientation.LEFT),
+            PlaquetteSide.DOWN,
+            measurement_basis=MeasurementBasis.X,
+        ),
+        2: xxxx_memory_plaquette(),
+        3: _with_measurements_on_data_qubits_on_side(
+            zzzz_memory_plaquette(),
+            PlaquetteSide.DOWN,
+            measurement_basis=MeasurementBasis.X,
+        ),
+        6: zzzz_memory_plaquette(),
+        7: xxxx_memory_plaquette(),
+        8: zz_memory_plaquette(PlaquetteOrientation.RIGHT),
+    }
+    return StandardComputationBlock(
+        QubitHorizontalBorders(dimension),
+        TemporalPlaquetteSequence(
+            initial_plaquettes, repeating_plaquettes, final_plaquettes
+        ),
+    )
+
+
+def xzo_block(dimension: LinearFunction) -> StandardComputationBlock:
+    xzz = xzz_block(dimension)
+    xzz.repeating_plaquettes = None
+    return xzz
+
+
+def zxo_block(dimension: LinearFunction) -> StandardComputationBlock:
+    zxz = zxz_block(dimension)
+    zxz.repeating_plaquettes = None
+    return zxz
```

Comment:
The functionality of time-direction connector blocks is: 
    1. replace the measurement layer of the cube beneath it with a memeory layer
    2. replace the initialization layer of the cube above it with a memory layer

So it seems this implementation is incorrect in the sense.

Reply:
I think the current implementation is still problematic. Maybe some changes on the `Computation` class need to be made to account for the action "remove the initialization/measurement layer of the cube it connects to".

Reply:
Yes, the methods are present (`StandardComputationBlock.replace_boundary_with_empty_plaquettes` and `StandardComputationBlock.instantiate_without_boundary`) but I wanted to wait for a graph-like structure in order to avoid computing which boundaries should be removed / replaced. I think this can be done in a follow-up PR, do you think I should rather implement everything here?

Reply:
I think it's fine to leave this with another PR. Then I have no problems and I'll approve the merge:)